### DISCUSSION TITLE
Depth Occlusion + VideoNode + Optimized Plane rendering

### DISCUSSION THREAD

**Discussion by ThomasGorisse** [sceneview/sceneform-android#88] [General]
- Depth Occlusion Rendering (#84):
    - To enable it:
        - `arSceneView.getCameraStream().setDepthOcclusionMode(CameraStream.DepthOcclusionMode.DEPTH_OCCLUSION_ENABLED);`
        - `if (session.isDepthModeSupported(Config.DepthMode.AUTOMATIC)) {config.setDepthMode(Config.DepthMode.AUTOMATIC);}`
    - Even when Depth is supported the user can disable the occlusion material
    - Added Depth usage for the gltf samples app
- Added VideoNode and updated video-texture sample. (#83):
    - Added VideoNode and updated videotexture sample accordingly.
    - Added option to pass own ExternalTexture to VideoNode.
- Optimized the Plane rendering (#78):
    - Prevents unnecessary calls to calculate the `focusPoint` and modify the `planeMaterial`.
    - Added a new `enum PlaneRendererMode` to render all detected planes or only the top most.
        - Choose `RENDER_ALL` to get the old behaviour. Everything what is visible to the user is rendered.
        - Choose `RENDER_TOP_MOST` to render only the top most plane.
    - Don't call the update function of the `PlaneRenderable` if it is not enabled.
    - Added `arFragment.getArSceneView().getPlaneRenderer().setEnabled(false);`
- Fixed RenderableInstance getMaterial()

<hr /><em>This discussion was created from the release <a href='https://github.com/ThomasGorisse/sceneform-android-sdk/releases/tag/v1.19.1'>Depth Occlusion + VideoNode + Optimized Plane rendering </a>.</em>

**Comment by ThomasGorisse** [comment]
A great thank to @RGregat, @Venthorus, @grassydragon for this big step forward release.
And @romainguy and @prideout for answering us on the Filament part.

**Reply by RGregat** [reply]
Awesome job!

**Reply by issacclee** [reply]
Great work guys!
